account/manager.go

```diff
@@ -107,6 +108,58 @@ func (wt witnessType) witnessSize() (int, error) {
 	}
 }
 
+// Action is a type of account modification.
+type Action string
+
+const (
+	CREATE   Action = "create"
+	DEPOSIT  Action = "deposit"
+	WITHDRAW Action = "withdraw"
+	RENEW    Action = "renew"
+	CLOSE    Action = "close"
+)
+
+type TxLabel struct {
+	Account AccountTxLabel `json:"account"`
+}
+
+type AccountTxLabel struct {
+	Key           string         `json:"key"`
+	Action        Action         `json:"action"`
+	ExpiryHeight  uint32         `json:"expiry_height"`
+	OutputIndex   uint32         `json:"output_index"`
+	IsExpirySpend bool           `json:"expiry_spend"`
+	TxFee         btcutil.Amount `json:"tx_fee"`
+	BalanceDiff   btcutil.Amount `json:"balance_diff"`
+}
+
+// actionTxLabel returns a transaction label for use with account
+// modification actions.
+func actionTxLabel(account *Account, action Action, isExpirySpend bool,
+	txFee btcutil.Amount, balanceDiff btcutil.Amount) (string, error) {
+
+	prefixTag := "poold -- %s"
+
+	acctKey := account.TraderKey.PubKey.SerializeCompressed()
+	key := fmt.Sprintf("%x", acctKey)
+	label := AccountTxLabel{
+		Key:           key,
+		Action:        action,
+		ExpiryHeight:  account.Expiry,
+		OutputIndex:   account.OutPoint.Index,
+		IsExpirySpend: isExpirySpend,
+		TxFee:         txFee,
+		BalanceDiff:   balanceDiff,
+	}
+	labelJson, err := json.Marshal(label)
+	if err != nil {
```

Comment:
Since this error should only happen when using the JSON API incorrectly, I think we could simplify this helper function by removing the error return type and instead just logging the error here and return a default label (e.g. `pool -- <action>`) as a fall back. That would remove some of the `if err != nil ` checks further down in the diff.

Reply:
Good idea!